responsemanager/responsemanager.go

```diff
@@ -148,23 +149,24 @@ type finishResponseRequest struct {
 }
 
 func (rm *ResponseManager) processQueriesWorker() {
+	const targetWork = 1
 	taskDataChan := make(chan *responseTaskData)
 	var taskData *responseTaskData
 	for {
-		nextTaskBlock := rm.queryQueue.PopBlock()
-		for nextTaskBlock == nil {
+		pid, tasks, _ := rm.queryQueue.PopTasks(targetWork)
```

Comment:
Third return value is "remaining work" for this peer, which we really don't care about.